tests/testAutopas/tests/autopasInterface/IteratorTest.cpp

```diff
@@ -453,6 +453,84 @@ TEST_P(IteratorTest, testOpenMPRegionIteratorsHaloOnly) {
   }
 }
 
+/**
+ * The main idea of this test is to check whether deletion through a RegionIterator doesn't do stupid things.
+ */
+template <bool testConstIterators>
+void testRegionIteratorDeletion(autopas::ContainerOption containerOption, double cellSizeFactor, bool priorForceCalc) {
+  // create AutoPas object
+  autopas::AutoPas<Molecule, FMCell> autoPas;
+  // Reference to the AutoPas object to be able to check const iterators.
+  std::conditional_t<testConstIterators, const autopas::AutoPas<Molecule, FMCell> &,
+                     autopas::AutoPas<Molecule, FMCell> &>
+      autoPasRef = autoPas;
+
+  autoPas.setAllowedContainers(std::set<autopas::ContainerOption>{containerOption});
+  autoPas.setAllowedCellSizeFactors(autopas::NumberSetFinite<double>(std::set<double>({cellSizeFactor})));
+
+  defaultInit(autoPas);
+
+  size_t numParticles = 200;
```

Comment:
do we really need to test 200 particles here?

Reply:
just for safety :) s.t. we really have multiple particles that are deleted and also particles after the deleted particles in the same towers/cells.